vendor/k8s.io/kubernetes/test/e2e/apimachinery/crd_publish_openapi.go

```diff
@@ -792,3 +776,186 @@ properties:
         type: array
         items:
           type: object`)
+
+func waitForCRDDefinitionInAllAPIServersPod(definitionName string, apiServers []string) *v1.Pod {
+	script := `
+       apk add curl jq
+       TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+       KEY="${DEFINITION_KEY}"
+       SEEN=0
+       PREV_HTTP_BODY=""
+       PREV_ETAG=""
+
+       while [ ${SEEN} -lt ${SERVERS_LEN} ]
+       do
+         for server in ${SERVERS}
+         do
+           URL="https://$server/openapi/v2"
+           echo "downloading the OpenAPI spec from ${URL}, prev_etag=${PREV_ETAG}, seen=${SEEN}"
+           HTTP_RESPONSE=$(curl -k -s -w "HTTPSTATUS:%{http_code}" -H "If-None-Match:\"${PREV_ETAG}\"" -H "Authorization: Bearer $TOKEN" --dump-header rsp-header.json ${URL})
+           HTTP_STATUS=$(echo $HTTP_RESPONSE | tr -d '\n' | sed -e 's/.*HTTPSTATUS://')
+           HTTP_BODY=$(echo $HTTP_RESPONSE | sed -e 's/HTTPSTATUS\:.*//g')
+           ETAG=$(cat rsp-header.json | grep etag: | sed -e 's/etag://' | sed -e 's/^ //' | tr -d '\r\n' | tr -d '"')
+           echo "parsing the response"
+           if [ $HTTP_STATUS -eq 304 ]; then
+             echo "StatusNotModified returned, rewriting the previous body and etag"
+             HTTP_BODY=${PREV_HTTP_BODY}
+             ETAG=${PREV_ETAG}
+           fi
+           echo ${HTTP_BODY} | jq -e ${KEY} > /dev/null
+           if [ $? -eq 0 ]; then
+             echo "found the key in the response"
+             PREV_HTTP_BODY=${HTTP_BODY}
+             PREV_ETAG=${ETAG}
+             SEEN=$(( SEEN + 1 ))
+           else
+            echo "haven't found the key=${KEY}, resetting the counter and trying one more time"
+            PREV_HTTP_BODY=""
+            PREV_ETAG=""
+            SEEN=0
+           fi
+         done
+       done
+`
+
+	r := strings.NewReplacer(
+		"${SERVERS_LEN}", fmt.Sprintf("%d", len(apiServers)),
+		"${SERVERS}", apiServersToString(apiServers),
+		"${DEFINITION_KEY}", fmt.Sprintf(".definitions.\\\"%s\\\"", definitionName),
+	)
+	script = r.Replace(script)
+
+	name := "wait-for-crd-definition"
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name + "-" + string(uuid.NewUUID()),
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    name,
+					Image:   "alpine",
```

Comment:
Is it okay to use an image like this one?
I couldn't find one (`imageutils.GetE2EImage`) that would have or allow me to install `curl, jq`

Reply:
No unfortuntaely ti is not.  You must use an image defined in the upstream utils/image package.

Reply:
If no existing upstream image supports jq you can't use it in your test until you get an image added, THEN you need to get that added upstream, THEN it needs to be part of upstream image publication.

Reply:
oh, okay, thanks for the info.

Reply:
@smarterclayton would you know how to add a new image?